src/common/meta/src/kv_backend/postgres.rs

```diff
@@ -0,0 +1,619 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::borrow::Cow;
+use std::sync::Arc;
+
+use snafu::ResultExt;
+use tokio_postgres::types::ToSql;
+use tokio_postgres::{Client, NoTls};
+
+use super::{KvBackend, TxnService};
+use crate::error::{ConnectPostgresSnafu, Error, PostgresFailedSnafu, Result, StrFromUtf8Snafu};
+use crate::kv_backend::txn::{Txn as KvTxn, TxnResponse as KvTxnResponse};
+use crate::kv_backend::KvBackendRef;
+use crate::rpc::store::{
+    BatchDeleteRequest, BatchDeleteResponse, BatchGetRequest, BatchGetResponse, BatchPutRequest,
+    BatchPutResponse, CompareAndPutRequest, CompareAndPutResponse, DeleteRangeRequest,
+    DeleteRangeResponse, PutRequest, PutResponse, RangeRequest, RangeResponse,
+};
+use crate::rpc::KeyValue;
+
+/// Posgres backend store for metasrv
+pub struct PgStore {
+    // TODO: Consider using sqlx crate.
+    client: Client,
+}
+
+const EMPTY: &[u8] = &[0];
+
+const METADKV_CREATION: &str =
+    "CREATE TABLE IF NOT EXISTS metakv(k varchar PRIMARY KEY, v varchar)";
+
+const FULL_TABLE_SCAN: &str = "SELECT k, v FROM metakv $1";
+
+const POINT_GET: &str = "SELECT k, v FROM metakv WHERE k = $1";
+
+const PREFIX_SCAN: &str = "SELECT k, v FROM metakv WHERE k LIKE $1";
```

Comment:
I think we should keep the order of keys.

Reply:
added order by key to all the range queries